lib/teiserver_web/router.ex

```diff
@@ -617,6 +621,11 @@ defmodule TeiserverWeb.Router do
     get("/tools/falist", ToolController, :falist)
     get("/tools/test_page", ToolController, :test_page)
 
+    if Application.compile_env(:teiserver, Teiserver)[:test_mode] do
```

Comment:
`compile_env` gets the value at compile time, and it's generally a bad idea. Because you need to recompile the app to change it. Also, this complicate matters when compiling the app from a different machine.

If you want to conditionally have these endpoints + template, a better way is to put that into the config files under `config/config.exs` set to false, and inside `config/test.exs` `config/dev.exs` put that to true.

But, tbh, I'm not sure all of this is warranted, requiring admin/server privilege to be able to create users manually may just be enough.

Reply:
I have removed that for now. I currently think relying on the admin/ server privilege should be sufficient.